minesweeper/use-cases/startGame/startGame.ts

```diff
@@ -0,0 +1,67 @@
+import { BoardModel } from '@minesweeper/domain/Board.model'
+import { BombModel } from '@minesweeper/domain/Bomb.model'
+import { CellModel } from '@minesweeper/domain/Cell.model'
+import { FlagModel } from '@minesweeper/domain/Flag.model'
+import { NeighborsBombsCounter } from '@minesweeper/domain/NeighborsBombsCounter.model'
+
+interface IStartGameProps {
+	rows: number
+	columns: number
+	bombs: number
+}
+
+type IBoardResponse = BoardModel
+
+type IStartGameUseCase = (props: IStartGameProps) => IBoardResponse
+
+/**
+ * start game base on initial props
+ */
+
+export const startGameUseCase: IStartGameUseCase = ({
+	bombs: bombsInput,
+	rows,
+	columns,
+}) => {
+	if (bombsInput < 1) {
+		throw new Error('bombs must be greater than 0')
+	}
+	if (rows < 2) {
+		throw new Error('rows must be greater than 1')
+	}
+	if (columns < 2) {
+		throw new Error('columns must be greater than 1')
+	}
+
+	const totalCells = columns * rows
+
+	const cells: CellModel[] = [...Array(totalCells)].map(() => ({
+		exposed: false,
+		position: [0, 0],
+	}))
+
+	let total = 0
+
+	for (let y = 0; y < columns; y++) {
+		for (let x = 0; x < rows; x++) {
+			cells[total].position = [x, y]
+			total++
+		}
+	}
+
+	const neighBorsBombsCounter: NeighborsBombsCounter[] = Array(totalCells)
+
+	let bombs: BombModel[] = Array(bombsInput)
```

Comment:
i think this can be const

Reply:
i think to be let for push new bomb in this array, you can see new implementation

Reply:
Ok but for pushing you can do it with a push or even by index and it doesn't affect to be a const

You can do ` array[5] = false  `in a `const` and it allows you to do that

Reply:
It's true 😄